src/r2mm/profiles/ProfilesClient.ts

```
@@ -0,0 +1,34 @@
+import Axios, { AxiosResponse } from 'axios';
+import R2Error from 'src/model/errors/R2Error';
+
+export default class ProfilesClient {
+    public static TSCreateAPI = 'https://thunderstore.io/api/experimental/legacyprofile/create/'
+    public static TSRetrieveAPI = (profileImportCode: string) => {return `https://thunderstore.io/api/experimental/legacyprofile/get/${profileImportCode}/`}
+    public static HerokuRetrieveAPI = (profileImportCode: string) => {return `https://r2modman-hastebin.herokuapp.com/raw/${profileImportCode}`}
+
+    public static async sendProfile(endpoint: string, payload: string): Promise<AxiosResponse | R2Error> {
+        try {
+            return await Axios.post(
+                endpoint,
+                payload,
+                {
+                    headers: {'Content-Type': 'application/octet-stream'}
+                }
+            )
+        } catch (e: any) {
+            if (Axios.isAxiosError(e)) {
+                if (e.response!.status == 429) {
+                    return new R2Error('Too many exports in a short period of time', ':PeepoShake:', 'Slow down there champ');
+                } else {
+                    return new R2Error('Failed to export profile', e.message, `${e.response!.status}`);
+                }
+            } else {
+                return new R2Error('Unexpected error', e.message, null);

```

And since we're throwing them, you can just throw the original error here if we don't do any special handling for it and let the call site deal with it.